### ISSUE TITLE
Key Request

### ISSUE THREAD

**Issue by MichaelLeeHobbs** [AJS-development/OpenAgar#23]
uuid: CkwGATwoA0dtyYjAVxTE

**Comment by Andrews54757** [comment]
Oh @MichaelLeeHobbs . Your back!!!

**Comment by Andrews54757** [comment]
234852798622

**Comment by Andrews54757** [comment]
Also, you should wait before downloading yet. Because I am about to post a new update that adds multiverse like configurations

**Comment by Andrews54757** [comment]
Update done.

**Comment by LegitSoulja** [comment]
Welcome back

**Comment by Andrews54757** [comment]
I don't think he wants to come back and do agar.io though @LegitSoulja .

**Comment by MichaelLeeHobbs** [comment]
Sorta, I got a side job for a lot of money that's keep me very busy.

**Comment by MichaelLeeHobbs** [comment]
I might knock out a few bugs ;) Think you guys made the right move dumping the old code.

**Comment by Andrews54757** [comment]
Thanks, I have been working really hard on this. I wrote the entire thing from scratch

**Comment by Andrews54757** [comment]
@MichaelLeeHobbs , However, I have a question. In this program, the software takes advantage of multiple cores in a computer, using processes. And everything works out well, AI, collisions, leaderboard, and mass checking is on a different process. I made the processes so that when it's parent is dead, it kills itself. But, sometimes. it doesn't, especially when I stop and start the program frequently. I was wondering why it does that.


Also, you said you can make AI evil. I think this program is perfect for it since the bots run in a separate process, sparing the main process from all the lag caused by the ai.

**Comment by Andrews54757** [comment]
And I got sick of people complaining to me about performance in OgarUL. So I abandoned it.

**Comment by MichaelLeeHobbs** [comment]
@Andrews54757 Yeap, I'll take a look at the AI. Honestly for a game like this it should take too much work.

**Comment by MichaelLeeHobbs** [comment]
@Andrews54757 Quick question, where is the code for the front end? Now that we have our own client, I'd like to see if we can setup some Admin controls on the front end.

**Comment by Andrews54757** [comment]
?

**Comment by LegitSoulja** [comment]
@MichaelLeeHobbs we don't have our own client YET. It's in development still.

And what type of admin panel/controls?

**Comment by Andrews54757** [comment]
@MichaelLeeHobbs I fixed the child process issue. And we were also thinking of a 3d client. Except it got ruled out after we learned Socket.io is unavailable in other languages.

**Comment by LegitSoulja** [comment]
@Andrews54757 that wasn't the case. It was due to that fact Unity Engine wasn't supported for chrome in the language(C#) that I was going to develop it in. I still could make it, that's if Unity Engine for (Unity Script) isn't needed, but I doubt it, so I decided not to waste my time and re-write the client from what we already have, and make it our owns. I'm thinking about building the site fully not just a client, but with accounts, user rankings & communications. I want it to feel different from any other site. 

I'm making the site where if we want to add more things, it can be installed as plugins. It'll keep things more organized, because there will be ALOT of things added on the site, and it will make future editing faster and easier.

I would look into WebGL aswell since you recommended it. I was going to use a game engine, instead of hand writing because I know we can make the game easier, as virtually making it.

But I don't know what to do, better decide quick on what we shall do.

- Use WebGL
- Rewrite the client and make it more faster, and efficient, and remove other things from it that's not necessary
- Try Unity again, and learn unity script syntax. (Btw, Unity supports socket.io, and could be introduced to RSON).
- Just continue with what im doing?
